ppl/zqd/apiserver/postgresdb.go

```diff
@@ -0,0 +1,80 @@
+package apiserver
+
+import (
+	"context"
+	"errors"
+
+	"github.com/brimsec/zq/api"
+	"github.com/brimsec/zq/ppl/zqd/postgres"
+	"github.com/brimsec/zq/zqe"
+	"github.com/go-pg/pg/v10"
+)
+
+type PostgresDB struct {
+	db *pg.DB
+}
+
+func OpenPostgresDB(ctx context.Context, conf postgres.Config) (*PostgresDB, error) {
+	db := pg.Connect(conf.ToPGOptions())
+	if err := db.Ping(ctx); err != nil {
+		return nil, err
+	}
+	return &PostgresDB{db}, nil
+}
+
+func (d *PostgresDB) CreateSpace(ctx context.Context, row SpaceRow) error {
+	if row.ID == "" {
+		return zqe.ErrInvalid("row must have an id")
+	}
+
```

Comment:
Could you verify that `row.ParentID` isn't set? In the future, we could try to unify CreateSpace & CreateSubspace here & in the filedb.

Reply:
Good idea.